File: wpimath/src/main/native/cpp/controller/proto/ElevatorFeedforwardProto.cpp
// Copyright (c) FIRST and other WPILib contributors.
// Open Source Software; you can modify and/or share it under the terms of
// the WPILib BSD license file in the root directory of this project.

#include "frc/controller/proto/ElevatorFeedforwardProto.h"

#include "controller.pb.h"

google::protobuf::Message* wpi::Protobuf<frc::ElevatorFeedforward>::New(
    google::protobuf::Arena* arena) {
  return google::protobuf::Arena::CreateMessage<
      wpi::proto::ProtobufElevatorFeedforward>(arena);
}

frc::ElevatorFeedforward wpi::Protobuf<frc::ElevatorFeedforward>::Unpack(
    const google::protobuf::Message& msg) {
  auto m = static_cast<const wpi::proto::ProtobufElevatorFeedforward*>(&msg);
  return frc::ElevatorFeedforward{
      m->ks(),
      m->kg(),
      m->kv(),
      m->ka(),
  };
}

void wpi::Protobuf<frc::ElevatorFeedforward>::Pack(
    google::protobuf::Message* msg, const frc::ElevatorFeedforward& value) {
  auto m = static_cast<wpi::proto::ProtobufElevatorFeedforward*>(msg);
  m->set_ks(value.ks.value());
  m->set_kg(value.kg.value());
  m->set_kv(value.kv.value());
  m->set_ka(value.ka.value());
}

File: wpimath/src/main/java/edu/wpi/first/math/controller/proto/ElevatorFeedforwardProto.java
// Copyright (c) FIRST and other WPILib contributors.
// Open Source Software; you can modify and/or share it under the terms of
// the WPILib BSD license file in the root directory of this project.

package org.wpilib.math.controller.proto;

import org.wpilib.math.controller.ElevatorFeedforward;
import org.wpilib.math.proto.Controller.ProtobufElevatorFeedforward;
import org.wpilib.util.protobuf.Protobuf;
import us.hebi.quickbuf.Descriptors.Descriptor;

public class ElevatorFeedforwardProto
    implements Protobuf<ElevatorFeedforward, ProtobufElevatorFeedforward> {
  @Override
  public Class<ElevatorFeedforward> getTypeClass() {
    return ElevatorFeedforward.class;
  }

  @Override
  public Descriptor getDescriptor() {
    return ProtobufElevatorFeedforward.getDescriptor();
  }

  @Override
  public ProtobufElevatorFeedforward createMessage() {
    return ProtobufElevatorFeedforward.newInstance();
  }

  @Override
  public ElevatorFeedforward unpack(ProtobufElevatorFeedforward msg) {
    return new ElevatorFeedforward(msg.getKs(), msg.getKg(), msg.getKv(), msg.getKa(), msg.getDt());
  }

  @Override
  public void pack(ProtobufElevatorFeedforward msg, ElevatorFeedforward value) {
    msg.setKs(value.getKs());
    msg.setKg(value.getKg());
    msg.setKv(value.getKv());
    msg.setKa(value.getKa());
    msg.setDt(value.getDt());
  }
}
